plugins/@grouparoo/google-sheets/src/lib/sheet-import/downloadAndRefreshFile.ts

```diff
@@ -0,0 +1,54 @@
+import { log } from "actionhero";
+import fs from "fs-extra";
+import { tmpdir } from "os";
+import { SimpleSourceOptions } from "@grouparoo/core";
+import Spreadsheet from "./Spreadsheet";
+import csvWriter from "csv-write-stream";
+
+export async function downloadAndRefreshFile(
+  sourceId: string,
+  appOptions: SimpleSourceOptions,
+  sourceOptions: SimpleSourceOptions
+) {
+  const sheet = new Spreadsheet(
+    appOptions,
+    sourceOptions.sheet_url?.toString()
+  );
+
+  let localDir = `${tmpdir()}/google-sheets-cache/${sourceId}`;
+  let localPath = `${localDir}/${sheet.docId}.csv`;
+
+  let toDownload = false;
+  if (!fs.existsSync(localPath)) {
+    toDownload = true;
+    fs.mkdirpSync(localDir);
+  } else {
+    const stats = fs.statSync(localPath);
+    const now = new Date();
+    if (now.getTime() - stats.birthtime.getTime() > 1000 * 60)
```

Comment:
Let's make `60*1000` to be a variable at the top of the file to be clear that its the `CSV_CACHE_MILLISECONDS` or something.